Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26974 Better named tensor error messages.
* #26968 Make named tensor implementations more robust
* **#26914 Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)**

Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17609136](https://our.internmc.facebook.com/intern/diff/D17609136)

zou3519 committed:
Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)

Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

nairbv commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -899,6 +899,22 @@ inferSqueezeGeometry(const Tensor &tensor) {
 }
 
+#ifdef BUILD_NAMEDTENSOR
+optional<std::vector<Dimname>> inferSqueezeNames(const Tensor& tensor) {
```
should this be in NamedTensorUtils.cpp?

zou3519 replied:
Sure

zou3519 committed:
Update on "Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)"


Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17609136](https://our.internmc.facebook.com/intern/diff/D17609136)

zou3519 committed:
Update on "Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)"


Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17609136](https://our.internmc.facebook.com/intern/diff/D17609136)

zou3519 committed:
Update on "Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)"


Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17609136](https://our.internmc.facebook.com/intern/diff/D17609136)

zou3519 committed:
Update on "Named tensor support for: index_fill_, index_fill, squeeze, median(Tensor)"


Also added dimname overloads for index_fill_ and squeeze.

Test Plan:
- [namedtensor ci]

Differential Revision: [D17609136](https://our.internmc.facebook.com/intern/diff/D17609136)
